Marlin/configuration_store.cpp

```diff
@@ -140,35 +144,51 @@ void _EEPROM_writeData(int &pos, uint8_t* value, uint8_t size) {
       SERIAL_ECHO_START;
       SERIAL_ECHOLNPGM(MSG_ERR_EEPROM_WRITE);
     }
+    eeprom_checksum += c;
     pos++;
     value++;
   };
 }
 void _EEPROM_readData(int &pos, uint8_t* value, uint8_t size) {
   do {
-    *value = eeprom_read_byte((unsigned char*)pos);
+    byte c = eeprom_read_byte((unsigned char*)pos);
+    *value = c;
+    eeprom_checksum += c;
     pos++;
     value++;
   } while (--size);
 }
 #define EEPROM_WRITE_VAR(pos, value) _EEPROM_writeData(pos, (uint8_t*)&value, sizeof(value))
 #define EEPROM_READ_VAR(pos, value) _EEPROM_readData(pos, (uint8_t*)&value, sizeof(value))
 
+#define DUMMY_PID_VALUE 3000.0f
+#define EEPROM_OFFSET 100
+
 /**
- * Store Configuration Settings - M500
+ * Post-process after Retrieve or Reset
  */
+void Config_Postprocess() {
+  // steps per s2 needs to be updated to agree with units per s2
+  planner.reset_acceleration_rates();
 
-#define DUMMY_PID_VALUE 3000.0f
+  #if ENABLED(DELTA)
+    recalc_delta_settings(delta_radius, delta_diagonal_rod);
+  #endif
 
-#define EEPROM_OFFSET 100
+  #if ENABLED(PIDTEMP)
+    thermalManager.updatePID();
+  #endif
+
+  calculate_volumetric_multipliers();
+}
 
 #if ENABLED(EEPROM_SETTINGS)
 
 /**
- * Store Configuration Settings - M500
+ * M500 - Store Configuration
  */
-
 void Config_StoreSettings()  {
+  eeprom_checksum = 0;
   float dummy = 0.0f;
   char ver[4] = "000";
   int i = EEPROM_OFFSET;
```

Comment:
EEPROM_WRITE_VAR(i, ver); // invalidate data first
 EEPROM_WRITE_VAR(i, eeprom_checksum ); // or i += 4
// Now eeprom_checksum = 0 for calc checksum
eeprom_checksum = 0;